app/main/views/suppliers.py

```diff
@@ -50,12 +51,16 @@
 def find_suppliers():
     if request.args.get("supplier_id"):
```

Comment:
You probably didn't design this, but the behaviour of this view is weird - specifying a `supplier_id` makes it suddenly ignore other filtering parameters... guess this might also be a limitation from the API.

Reply:
Yeah. To be honest I'd prefer a separate view for that ID search, with an explicit url pattern looking for `/supplier/<supplier_id>`. We might look at that when we do the 'display supplier info' page work for the CCS data user as that page is going to need a new view anyway. I'll keep it here for now.